modules/nw-ingress-controller-nodeportservice-projects.adoc

```diff
@@ -0,0 +1,147 @@
+
+.Prerequisites
+
+* You installed the {oc-first}.
+* Logged in as a user with `cluster-admin` privileges.
+* You created a wildcard DNS record.
+// https://docs.openshift.com/container-platform/4.16/networking/ingress-controller-dnsmgt.html (does not detail how to create the DNS)
+
+.Procedure
+
+. Create a custom resource (CR) file for the Ingress Controller:
++
+.Example of a CR file that defines information for the `IngressController` object
+[source,yaml]
+----
+apiVersion: v1
+items:
+- apiVersion: operator.openshift.io/v1
+  kind: IngressController
+  metadata:
+    name: <custom_name> <1>
+    namespace: openshift-ingress-operator
+  spec:
+    replicas: 1
+    domain: <domain> <2>
+    nodePlacement:
+      nodeSelector:
+        matchLabels:
+          ingresscontroller: <custom_name> <3>
+    namespaceSelector:
+     matchLabels:
+       ic: <custom_name> <3>
+    endpointPublishingStrategy:
+      type: NodePortService 
+kind: List
+metadata:
+  resourceVersion: ""
+  selfLink: ""
+# ...
+----
+<1> Specify the a custom `name` for the `IngressController` CR.
+<2> The DNS name that the Ingress Controller services.
+<3> Specify the label for a set of namespaces.
+
+. Add a label to a node by using the `oc label node` command:
++
+[source,terminal]
+----
+$ oc label node <node_name> ingresscontroller=<ingress_controller_custom_resource> <1>
+----
+<1> Where `<ingress_controller_custom_resource>` must match the value that you specified in the `nodePlacement` section of your `IngressController` CR.
+
+. Create the `IngressController` object:
++
+[source,terminal]
+----
+$ oc create -f <ingress_controller_cr>.yaml
+----
+
+. Find the port for the service that was created for the `IngressController` CR:
++
+[source,terminal]
+----
+$ oc get svc -n openshift-ingress
+----
++
+.Example output that shows port `80:32432/TCP` for the `router-nodeport-custom-ic3` service
+[source,terminal]
+----
+NAME                        TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)                                     AGE
+router-internal-default     ClusterIP   172.30.195.74    <none>        80/TCP,443/TCP,1936/TCP                     223d
+router-nodeport-custom-ic3   NodePort    172.30.109.219   <none>        80:32432/TCP,443:31366/TCP,1936:30499/TCP   155m
+----
+
+. Use the `oc new-project` command to create a new project:
++
+[source,terminal]
+----
+$ oc new-project <ingress_controller_name>
+----
+
+. Enter the following command to label the additional Ingress Controller:
++
+[source,terminal]
+----
+$ oc label namespace <ingress_controller_name> ic=<ingress_controller_name> <1>
+----
+<1> Where `<ingress_controller_name>` must match the value that you specified in the `namespaceSelector` section of your Ingress Controller CR.
+
+. Create a new application in your cluster:
++
+[source,terminal]
+----
+$ oc new-app --image=quay.io/openshifttest/<service_name>:multiarch
```

Comment:
we could go two ways with this:
```suggestion
$ oc new-app --image=<image name>
```
OR use a real example:
```suggestion
$ oc new-app --image=quay.io/openshifttest/hello-openshift:multiarch
```

Reply:
:heavy_check_mark: